paasta_tools/deployd/watchers.py

```diff
@@ -23,6 +26,59 @@ def __init__(self, inbox_q, cluster):
         self.is_ready = False
 
 
+class AutoscalerWatcher(PaastaWatcher):
+
+    def __init__(self, inbox_q, cluster, **kwargs):
+        super(AutoscalerWatcher, self).__init__(inbox_q, cluster)
+        self.zk = kwargs.pop('zookeeper_client')
+        self.watchers = {}
+
+    def watch_folder(self, path):
+        """recursive nonsense"""
+        if "autoscaling.lock" in path:
+            return
+        self.log.debug("Adding watch on {}".format(path))
+        watcher = ChildrenWatch(self.zk, path, func=self.process_folder_event, send_event=True)
+        self.watchers[path] = watcher
+        children = watcher._prior_children
+        if children and ('instances' in children):
+            self.watch_node("{}/instances".format(path))
+        elif children:
+            for child in children:
+                self.watch_folder("{}/{}".format(path, child))
+
+    def watch_node(self, path):
+        self.log.debug("Adding watch on {}".format(path))
+        DataWatch(self.zk, path, func=self.process_node_event, send_event=True)
+
+    def process_node_event(self, data, stat, event):
+        self.log.debug("Node change: {}".format(event))
+        if event and (event.type == EventType.CREATED or event.type == EventType.CHANGED):
+            service, instance = event.path.split('/')[-3:-1]
+            self.log.info("Number of instances changed or autoscaling enabled for first time"
+                          " for {}.{}".format(service, instance))
+            service_instance = ServiceInstance(service=service,
+                                               instance=instance,
+                                               bounce_by=int(time.time()),
+                                               bounce_timers=None,
+                                               watcher=self.__class__.__name__)
+            self.inbox_q.put(service_instance)
+
+    def process_folder_event(self, children, event):
+        self.log.debug("Folder change: {}".format(event))
+        if event and (event.type == EventType.CHILD):
+            fq_children = ["{}/{}".format(event.path, child) for child in children]
+            for child in fq_children:
+                if child not in self.watchers:
+                    self.watch_folder(child)
+
+    def run(self):
+        self.watchers['/autoscaling'] = self.watch_folder('/autoscaling')
```

Comment:
I'm worried that you are "hard-coding" these folders here, and I'm pretty sure they are wrong. (where is the cluster name?)

Please use (and refactor as needed) the function that returns this path
https://github.com/Yelp/paasta/blob/master/paasta_tools/long_running_service_tools.py#L322

Wow ok mabe the zk client itself is already chrooted inside the cluster-specific namespace